x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -276,6 +294,14 @@ private boolean isConcreteRestrictedIndex(String indexPattern) {
         return RestrictedIndicesNames.isRestricted(indexPattern);
     }
 
+    private static boolean authorizeMappingUpdateBwcSpecialCase(Group group, String action) {
+        return action.equals(PutMappingAction.NAME) ||
```

Comment:
I think we'll need `AutoPutMapping` too.

Reply:
Yes, as we've discussed on another channel, `create_doc`, `create`, `index` lose all mapping update privileges, and `write` only retains the auto put mapping privilege. For bwc reasons, put mapping and auto put mapping are still allowed on indices and aliases, but not on data streams.